internal/core/services/rules_test.go

```diff
@@ -12,3 +12,41 @@ func TestDoesServiceMatchRule(t *testing.T) {
 		assert.False(t, DoesServiceMatchRule(endpoint, "== ++ abc 1jj +"))
 	})
 }
+
+func TestMapFunctions(t *testing.T) {
+	interfacemap := map[interface{}]interface{}{"hello": "world", "good": "bye"}
+	stringmap := map[string]interface{}{"hello": "world", "good": "bye"}
+	t.Run("Get()", func(t *testing.T) {
+		var val interface{}
+		var err error
+		_, err = ruleFunctions["Get"](stringmap, "hello")
+		assert.Error(t, err, "should error out when wrong map type is used")
+
+		_, err = ruleFunctions["Get"](interfacemap, 3)
+		assert.Error(t, err, "should error out when wrong key type is used")
```

Comment:
This is a separate test case.  You can still share the common fixture data but still use separate `t.Run` invocations to break these down properly.